sequencer/src/api/endpoints.rs

```diff
@@ -121,3 +124,52 @@ where
 
     Ok(api)
 }
+
+pub(super) fn state_signature<N, S>() -> anyhow::Result<Api<S, Error>>
+where
+    N: network::Type,
+    S: 'static + Send + Sync + ReadState,
+    S::State: Send + Sync + StateSignatureDataSource<N>,
+{
+    let toml = toml::from_str::<toml::Value>(include_str!("../../api/state_signature.toml"))?;
+    let mut api = Api::<S, Error>::new(toml)?;
+
+    api.get("getstatesignature", |req, state| {
+        async move {
+            let height = req
+                .integer_param("height")
+                .map_err(|err| Error::internal(err.to_string()))?;
```

Comment:
Can you just do `?` here without the `map_err`? I think `Error` should have a `From<RequestError>` impl, and we want this to return 400 something when there's a bad request, not 500 for internal error

If `?` doesn't work on it's own, you could also try [`from_request_error`](https://tide-disco.docs.espressosys.com/tide_disco/error/trait.Error.html#method.from_request_error)

Reply:
But the `Error` here is `hotshot_query_service::Error`.

Reply:
Yes but `?` does an implicit `.into()`. And `hotshot_query_service::Error` implements `tide_disco::Error` which provides `from_request_error`

Reply:
using `from_request_error` in [2f4ab54](https://github.com/EspressoSystems/espresso-sequencer/pull/879/commits/2f4ab5484d02c57dbfc0dca96ca4b183d5f355d6)